corehq/apps/accounting/management/commands/list_customer_billing_account_software_plan.py

```diff
@@ -0,0 +1,98 @@
+from django.core.management.base import BaseCommand
+from corehq.apps.accounting.models import BillingAccount, Subscription, SubscriptionType
+import csv
+
+
+class Command(BaseCommand):
+    # Always run this command without '--update' first
+    # let Ops review the output before running with '--update'
+    help = 'List customer billing accounts and associated software plans'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--update', action='store_true', default=False,
+                            help='Update all software plans (type PRODUCT or IMPLEMENTATION) '
+                            'under each billing account to the main billing subscription')
+
+    def handle(self, *args, **kwargs):
+        update = kwargs['update']
+        billing_account_names = []
+        while True:
+            account_name = input("Enter a customer billing account name, "
+                                 "type 'ALL' for all customer billing account, "
+                                 "type 'DONE' when finished:\n")
+            if account_name.upper() == 'ALL':
+                billing_account_names = ['ALL']
+                break
+            elif account_name.upper() == 'DONE':
+                break
+            else:
+                try:
+                    BillingAccount.objects.get(name=account_name, is_active=True, is_customer_billing_account=True)
```

Comment:
What do you think of saving the value returned here rather than doing another query at the end to lookup all entered names?

Reply:
Yes, this will improve the performance. Fixed here: fadda1231b19693fefad5af269f5a7a05035680f